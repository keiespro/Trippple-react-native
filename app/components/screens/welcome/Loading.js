import React from 'react'
import {View, ActivityIndicator, Dimensions} from 'react-native'
import colors from '../../../utils/colors'

const DeviceHeight = Dimensions.get('window').height
const DeviceWidth = Dimensions.get('window').width


const Loading = () => (
  <View
    style={{
      justifyContent: 'center',
      alignItems: 'center',
      flexGrow: 1,
      position: 'absolute',
      top: 0,
      left: 0,
      width: DeviceWidth,
      height: DeviceHeight,
      backgroundColor: colors.outerSpace
    }}
  >
    <ActivityIndicator
      size="large"
      color={colors.white}
      animating
      style={{}}
    />
  </View>
)

export default Loading
