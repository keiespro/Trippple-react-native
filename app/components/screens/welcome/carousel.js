/*
* @noflow
*/

import {
  StyleSheet,
  Text,
  View,
  Image,
  ScrollView,
  Dimensions,
} from 'react-native';
import React, {Component} from 'react';


import colors from '../../../utils/colors'

import {MagicNumbers} from '../../../utils/DeviceConfig'

const DeviceHeight = Dimensions.get('window').height
const DeviceWidth = Dimensions.get('window').width

// import dismissKeysboard from 'dismissKeyboard'

const theslides = [
  {
    title: ' ',
    img: require('./assets/logo.png'),
    content: ' '
  },
  {
    title: 'BROWSE',
    img: require('./assets/tour-browse.png'),
    content: 'Find like-minded Couples and Singles.'
  },
  {
    title: 'MATCH',
    img: require('./assets/tour-match.png'),
    content: 'If they like you too, we\'ll connect you.'
  },
  {
    title: 'CONNECT',
    img: require('./assets/tour-connect.png'),
    content: 'Chat with real Couples or Singles who share your interests.'
  },
  {
    title: 'PRIVATE & DISCREET',
    img: require('./assets/tour-privacy.png'),
    content: 'Protect your identity. Easily block friends and family.'
  },

];

class Carousel extends Component {
  constructor() {
    super()
    this.state = {
      slides: theslides,
      offset: 0,
      index: 0
    }
  }
  handleScroll(e){
    let off = e.nativeEvent.contentOffset.x;
    if(off % DeviceWidth > 0) return;
    if(off < this.state.offset) {
      if(off < 0) {
        off = e.nativeEvent.contentOffset.x + (DeviceWidth * 5)
      }
    }else if(off >= DeviceWidth * this.state.slides.length){
      off = e.nativeEvent.contentOffset.x - (DeviceWidth * 5)
    }
    this.setState({
      offset: off,
      index: parseInt(off / DeviceWidth)
    })
  }
  render() {
    const {slides} = this.state
    const l = slides.length
    const welcomeSlides = [...slides, ...slides].map((slide, i) => (
      <View
        key={`${i}slide${i % l}`}
        style={[styles.slide]}
      >
        <Image
          style={{
            marginBottom: 25,
            height: MagicNumbers.is4s ? 150 : (DeviceHeight / 3) + MagicNumbers.screenPadding,
            paddingTop: 0,
            marginTop: i % l == 0 ? 100 : 0, // i % l == 0 ? MagicNumbers.screenPadding * 1.8 : MagicNumbers.screenPadding,
            width: i % l == 0 ? MagicNumbers.screenWidth + 50 : MagicNumbers.screenPadding * 4
          }}
          source={slide.img}
          defaultSource={slide.img}
          resizeMode={Image.resizeMode.contain}
        />
        <View style={[styles.textwrap, {marginBottom: 5}]} >
          <Text
            style={[styles.textplain, {
              fontFamily: 'montserrat',
              fontWeight: '800',
              marginTop: 15,
              fontSize: MagicNumbers.is4s ? 18 : 22,
            }]}
          >{slide.title}</Text>
        </View>
        {slide.content && (
          <View style={styles.textwrap}>
            <Text
              style={[styles.textplain, {
                fontSize: MagicNumbers.is5orless ? 18 : 20,
              }]}
            >{slide.content}</Text>
          </View>
        )}
      </View>
    ));

    return (
      <View collapsable>
        <ScrollView
          onScroll={this.handleScroll.bind(this)}
          style={[styles.carousel, { height: DeviceHeight - 140}]}
          horizontal
          pagingEnabled
          scrollEventThrottle={DeviceWidth * 5}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          scrollsToTop={false}
          contentOffset={{x: this.state.offset, y: 0}}
          removeClippedSubviews
          grayDots
          contentContainerStyle={{
            alignItems: 'center',
            justifyContent: 'center',
            alignSelf: 'stretch',
          }}
        >
          {welcomeSlides}
        </ScrollView>

        <View
          pointerEvents="none"
          style={{
            position: 'absolute',
            bottom: MagicNumbers.screenPadding / 2 - 10,
            left: 0,
            right: 0,
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center'
          }}
        >
          {this.state.slides.map((s, i) => <View key={`${i}dots`} style={[styles.dot, this.state.index == i ? styles.activeDot : null]} />)}
        </View>
      </View>

    )
  }
}

Carousel.displayName = 'Carousel';

export default Carousel

const styles = StyleSheet.create({
  dot: {
    backgroundColor: colors.shuttleGray,
    width: 12,
    height: 12,
    borderRadius: 6,
    marginLeft: 4,
    borderWidth: 2,

    marginRight: 4,
    marginTop: 2,
    marginBottom: 2,
    borderColor: colors.shuttleGray
  },
  activeDot: {
    backgroundColor: colors.mediumPurple,
    marginLeft: 4,
    marginRight: 4,
    marginTop: 2,
    marginBottom: 2,
    width: 12,
    height: 12,
    borderWidth: 2,
    borderColor: colors.mediumPurple
  },
  textplain: {
    color: colors.white,
    alignSelf: 'center',
    fontSize: 22,
    fontFamily: 'omnes',
    textAlign: 'center'
  },
  buttonText: {
    fontSize: 22,
    color: colors.white,
    alignSelf: 'center',
    fontFamily: 'montserrat'
  },
  carousel: {
    marginTop: 0,
    width: DeviceWidth,
    // height: DeviceHeight - 190,

  },
  slide: {
    width: DeviceWidth,
    flexDirection: 'column',
    // height: DeviceHeight - 190,
    justifyContent: 'flex-start',
    alignItems: 'center',
    padding: MagicNumbers.screenPadding / 2
  },


  textwrap: {
    alignItems: 'center',
    height: 50,
    justifyContent: 'center',
  },

  // dot: {
  //   backgroundColor: colors.shuttleGray,
  //   width: 16,
  //   height: 16,
  //   borderRadius: 8,
  //   marginLeft: 8,
  //   marginRight: 8,
  //   marginTop: 3,
  //   marginBottom: 3,
  //   borderColor: colors.shuttleGray
  // },
  // activeDot: {
  //   backgroundColor: colors.mediumPurple20,
  //   width: 16,
  //   height: 16,
  //   borderRadius: 8,
  //   marginLeft: 8,
  //   marginRight: 8,
  //   marginTop: 3,
  //   marginBottom: 3,
  //   borderWidth: 2,
  //   borderColor: colors.mediumPurple
  // }
});
