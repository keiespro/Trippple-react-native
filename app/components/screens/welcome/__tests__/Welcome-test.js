import 'react-native';
import React from 'react';
import {Welcome} from '../welcome';

import renderer from 'react-test-renderer';


const mockFn = jest.fn();

const Analytics = new mockFn()

jest.mock('react-native-device-info');

jest.mock('Geolocation');

jest.setMock('DeviceInfo', () => require.requireMock('DeviceInfo'));

test('renders correctly', () => {
  const tree = renderer.create(
    <Welcome />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
