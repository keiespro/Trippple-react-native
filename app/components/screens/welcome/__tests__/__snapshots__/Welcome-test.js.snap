// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "alignSelf": "stretch",
        "backgroundColor": "rgb(44,57,65)",
        "height": 1334,
        "justifyContent": "space-between",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": 750,
      },
    ]
  }
>
  <View>
    <View
      collapsable={true}
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "justifyContent": "center",
          }
        }
        contentOffset={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        grayDots={true}
        horizontal={true}
        onScroll={[Function]}
        pagingEnabled={true}
        removeClippedSubviews={true}
        scrollEventThrottle={3750}
        scrollsToTop={false}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        style={
          Array [
            Object {
              "marginTop": 0,
              "width": 750,
            },
            Object {
              "height": 1194,
            },
          ]
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 100,
                  "paddingTop": 0,
                  "width": 750,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                 
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                 
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 0,
                  "paddingTop": 0,
                  "width": 200,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                BROWSE
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                Find like-minded Couples and Singles.
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 0,
                  "paddingTop": 0,
                  "width": 200,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                MATCH
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                If they like you too, we'll connect you.
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 0,
                  "paddingTop": 0,
                  "width": 200,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                CONNECT
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                Chat with real Couples or Singles who share your interests.
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 0,
                  "paddingTop": 0,
                  "width": 200,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                PRIVATE & DISCREET
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                Protect your identity. Easily block friends and family.
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 100,
                  "paddingTop": 0,
                  "width": 750,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                 
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                 
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 0,
                  "paddingTop": 0,
                  "width": 200,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                BROWSE
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                Find like-minded Couples and Singles.
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 0,
                  "paddingTop": 0,
                  "width": 200,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                MATCH
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                If they like you too, we'll connect you.
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 0,
                  "paddingTop": 0,
                  "width": 200,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                CONNECT
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                Chat with real Couples or Singles who share your interests.
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "padding": 25,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              defaultSource={1}
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 494.6666666666667,
                  "marginBottom": 25,
                  "marginTop": 0,
                  "paddingTop": 0,
                  "width": 200,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "justifyContent": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "montserrat",
                      "fontSize": 22,
                      "fontWeight": "800",
                      "marginTop": 15,
                    },
                  ]
                }
              >
                PRIVATE & DISCREET
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "rgb(255,255,255)",
                      "fontFamily": "omnes",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                Protect your identity. Easily block friends and family.
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "bottom": 15,
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(86,96,103)",
                "borderColor": "rgb(86,96,103)",
                "borderRadius": 6,
                "borderWidth": 2,
                "height": 12,
                "marginBottom": 2,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 2,
                "width": 12,
              },
              Object {
                "backgroundColor": "rgb(132, 89, 218)",
                "borderColor": "rgb(132, 89, 218)",
                "borderWidth": 2,
                "height": 12,
                "marginBottom": 2,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 2,
                "width": 12,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(86,96,103)",
                "borderColor": "rgb(86,96,103)",
                "borderRadius": 6,
                "borderWidth": 2,
                "height": 12,
                "marginBottom": 2,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 2,
                "width": 12,
              },
              null,
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(86,96,103)",
                "borderColor": "rgb(86,96,103)",
                "borderRadius": 6,
                "borderWidth": 2,
                "height": 12,
                "marginBottom": 2,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 2,
                "width": 12,
              },
              null,
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(86,96,103)",
                "borderColor": "rgb(86,96,103)",
                "borderRadius": 6,
                "borderWidth": 2,
                "height": 12,
                "marginBottom": 2,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 2,
                "width": 12,
              },
              null,
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(86,96,103)",
                "borderColor": "rgb(86,96,103)",
                "borderRadius": 6,
                "borderWidth": 2,
                "height": 12,
                "marginBottom": 2,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 2,
                "width": 12,
              },
              null,
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "height": 140,
        "justifyContent": "space-around",
        "marginBottom": 20,
        "width": undefined,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "rgba(63, 115, 219, 1.00)",
                "flexDirection": "row",
                "flexGrow": 0,
                "height": 70,
                "justifyContent": "center",
                "margin": 0,
                "paddingHorizontal": 30,
              },
            ],
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 40,
              },
              Object {
                "marginRight": 5,
              },
              Object {
                "fontFamily": "EvilIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontFamily": "montserrat",
              "fontSize": 17,
              "fontWeight": "800",
            }
          }
        >
          LOG IN WITH FACEBOOK
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 30,
            "marginVertical": 20,
            "opacity": 1,
            "width": 100,
            "zIndex": 9999,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgb(120,129,135)",
                "fontFamily": "omnes",
                "fontSize": 12,
                "textAlign": "right",
                "textDecorationLine": "underline",
              }
            }
          >
            Why Facebook?
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 30,
            "marginVertical": 20,
            "width": 10,
            "zIndex": 9999,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgb(120,129,135)",
              "fontFamily": "omnes",
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          •
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 30,
            "marginVertical": 20,
            "opacity": 1,
            "width": 50,
            "zIndex": 9999,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgb(120,129,135)",
                "fontFamily": "omnes",
                "fontSize": 12,
                "textAlign": "left",
                "textDecorationLine": "underline",
              }
            }
          >
            Terms
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
